### PR TITLE
Fix docker arguments and load emba as volume into container

### PR DESCRIPTION
**Pull request by p4cx**
Now you don't have to rebuild the emba image for every small update in the code. Also all arguments are now working fine with docker. Have fun!

### CODE REVIEW

**Comment by m-1-k-3** [comment]
I like this idea for developping emba and testing the docker stuff. But it looks a bit risky to expose the complete emba directory writable to the docker container. For testing and dev this is fine but for massive scanning I do not think this should be exposed writable by default.

I think there are two possibilities we have:
First:
* -D 0 -> exposes emba not to the container and uses the emba that is already installed in the docker container -> should be the default beahaviour
* -D 1 -> exposes emba directory (for dev) -> the new stuff

Second:
We could try using readonly flags (read_only: true) in docker compose: https://stackoverflow.com/a/55760596

What do you think?

**Comment by p4cx** [comment]
Your argument seems to be valid, but then we have another flaw in our setup: We are now writing the extracted firmware and log files to the host as well. Could this be backfiring for the host? 
Which of these two options is your favourite? I will look further into this in the next few days and update this PR.

**Comment by m-1-k-3** [comment]
But we are not executing any code from log or from the extracted firmware afterwards on the host. So, this should be fine.

**Comment by p4cx** [comment]
`mkdir "/emba/test"` is resulting in `mkdir: cannot create directory '/emba/test': Read-only file system` - it should be save now.

File: helpers/log_folder.sh [hunk 1]
```diff
@@ -47,4 +47,24 @@ log_folder()
         ;;
     esac
   fi
+
+  readarray -t D_LOG_FILES < <( find . \( -path ./external -o -path ./config \) -prune -false -o \( -name "*.txt" -o -name "*.log" \) -exec md5sum {} \; 2>/dev/null | sort -u -k1,1 | cut -d\  -f3 )
```
Comment (m-1-k-3) [inline helpers/log_folder.sh:51]: could we move this to the prepare helper file? It probably fits there quite fine

**Review by m-1-k-3 — CHANGES_REQUESTED** [review]
code looks good. could you move this log_folder thing to the prepare helper?

File: helpers/path.sh [hunk 2]
```diff
@@ -18,6 +18,16 @@
 #                 firmware root path via $FIRMWARE_PATH
 #                 binary array via ${BINARIES[@]}
 
+check_path_valid() {
+  C_PATH="$1"
+  if [[ -n "$C_PATH" ]] && ( [[ "${C_PATH:0:2}" != "./" ]] && [[ "${C_PATH:0:2}" != "./" ]] && [[ "${C_PATH:0:3}" != "../" ]] ) ; then
+    print_output "[!] ""$C_PATH"" ""${C_PATH:0:1}"" ""${C_PATH:0:2}"" ""${C_PATH:0:3}"" "" is not a valid path in the context of emba" "no_log"
```
Comment (m-1-k-3) [inline helpers/path.sh]: I think we need to modify this check:

`  if [[ -n "$C_PATH" ]] && ( [[ "${C_PATH:0:1}" != "/" ]] && [[ "${C_PATH:0:2}" != "./" ]] && [[ "${C_PATH:0:2}" != "./" ]] && [[ "${C_PATH:0:3}" != "../" ]] ) ; then
`

Comment (p4cx) [reply helpers/path.sh]: Yes, you are right - it's done.

File: helpers/log_folder.sh (hunk 1 above)
Comment (p4cx) [reply helpers/log_folder.sh]: Done.

**Review by m-1-k-3 — APPROVED** [review]
looks good ... thank you. This will simplify testing and using docker a lot

**Pull request merged**